### DISCUSSION TITLE
Upgrading/Updating from V4.5 to V6

### DISCUSSION THREAD

**Discussion by FazzaGBR2** [snipe/snipe-it#11632] [General]
We have a very old version of Snipe-IT (V4.5) sitting on an equally old version of CentOS 7.

If we were to install the latest version of CentOS and then install the latest version of Snipe-IT can we simply export the database from within Snipe-IT itself on our old system and then import it in to the new one or has the structure of the data changed between V4.5 and V6 so that wont work?

Thanks.

**Comment by FazzaGBR2** [comment]
I now have another server with Snipe-IT V6 installed on it.  I have saved the Backup from V4.5, uploaded it to V6 but when I click to restore it, it logs me off and presents me with a setup screen where I have to create a new user - then it goes to the dashboard which is still empty so it has not imported any of our assets.

**Reply by nicokempe** [reply]
I have the exact same issue.

**Comment by FazzaGBR2** [comment]
Unfortunately the old 4.5 version of Snipe-IT we have was installed badly so when you try and run **php upgrade.php** (which I believe should upgrade the system to the latest version of Snipe-IT) it just fails with so many errors.   I'm not a Linux expert so have no idea where to even begin fixing them unfortunately.

**Comment by FazzaGBR2** [comment]
I've changed a load of permissions and a lot of the errors from running the **php upgrade.php** command have gone away but it wont update as it now says

"This version of PHP (7.1.29) is not compatible with Snipe-IT.  Snipe-IT requires PHP version 7.4.0 or greater.  Please upgrade your version of PHP"

I'm now trying to find out how to upgrade PHP to at least 7.4 but I'm going around in circles as everything a website is telling me to do just creates another error message for me to try and work out!! :(

**Reply by nicokempe** [reply]
here are some explanations: https://dev.to/anandpushkar088/upgrading-to-php-7-4-26dg

**Reply by FazzaGBR2** [reply]
Thanks.  I've now got PHP 7.4 installed and the **upgrade.php** script seems to run but finishes on Nothing To Migrate so Snipe-IT is still on V4.5 :(

I was hoping to upgrade/update the older installation to V6, perform a backup and download it and then upload the back to my new server with V6 on and do a restore and see if that works but I'm still stuck on V4.5.

**Reply by nicokempe** [reply]
yes i guess snipe-it changed something i am dealing with the same problem at the moment

**Reply by FazzaGBR2** [reply]
I've managed to run a Snipe-IT script to manually import the .zip file backup that I got from V4.5 and when I look in the actual SQL database I can see there is now some data in there!  I ran the **artisan snipeit:restore** command and referenced my downloaded zip file and that seemed to do the manual import.

My new/latest problem with all of this is now I am getting Not Found errors in my web browser saying HTTP Error 404. The requested resource is not found.

I must need to do something else now I've managed to get my data in the database but I'm not sure what so the fun continues!...

**Comment by FazzaGBR2** [comment]
I've spent over 30 hours trying to upgrade/update from V4.5 to V6 and it's just impossible as no one knows how to do it.

All I get now in a web browser is "Not Found" and "HTTP Error 404. The requested resource is not found."
